### PR TITLE
[SPARK-35397][SQL] Replace sys.err usage with explicit exception type

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This patch replaces `sys.err` usages with explicit exception types.

### Why are the changes needed?

Motivated by the previous comment https://github.com/apache/spark/pull/32519#discussion_r630787080, it sounds better to replace `sys.err` usages with explicit exception type.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala [hunk 1]
```diff
@@ -3338,7 +3338,8 @@ class Analyzer(override val catalogManager: CatalogManager)
       case _ : InnerLike =>
         (leftKeys ++ lUniqueOutput ++ rUniqueOutput, rightKeys)
       case _ =>
-        sys.error("Unsupported natural join type " + joinType)
+        throw QueryExecutionErrors.unsupportedNaturalJoinTypeError(joinType)
+
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala:3342]: nit. extra space.

**Review by dongjoon-hyun — APPROVED** [review]
Looks fine and consistent to me.

**Review by maropu — APPROVED** [review]

**Review by srowen — APPROVED** [review]
It will change the exception type, but, the exception generated now is uselessly broad anyway. I think it's fine.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala (hunk 1 above)
Comment (viirya) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala]: Removed. Thanks.

**Comment by dongjoon-hyun** [comment]
Merged to master for Apache Spark 3.2.0.
Thank you, @viirya and all!

**Comment by viirya** [comment]
Thank you @dongjoon-hyun @maropu @srowen

**Pull request closed without merging**